### PR TITLE
Fix: Make actions in userhub clickable

### LINKED ISSUE #2216: Streaming Payments API: Cancelling streams using motions
Parent issue: https://github.com/JoinColony/colonyCDapp/issues/2209

- New saga needs to be implemented
- Relevant block-ingestor handlers need to be implemented
- A corresponding colony action should be created
- It should be possible to annotate the transaction

### PR DESCRIPTION
**Pull request by davecreaser**
## Description

- Make items in the userhub clickable, to open the relevant action in the action sidebar
  - Stake items
  - Transaction items (if they have a relevant action hash)
- Update the UI of the userhub

Details here: https://github.com/JoinColony/colonyCDapp/issues/2261

## Testing

Check that the UI matches the designs: https://www.figma.com/design/l1dOM5qiQYwF0ElvKDqqjg/Design-System---Colony-v3?node-id=1782-50890&t=Zn39wYmkuOXiZ0z7-0

Check that stake items in the userhub are clickable, and open the action sidebar with the correct action.

Check that transaction items open the action sidebar with the correct action ONLY IF they have a hash which has a relevant action.

## Diffs

**Changes** 🏗

- Add buttons to the userhub items to create links to the relevant actions (transactions can now be expanded using the chevron button)
- Change html and classes of the userhub items to match the new designs

Resolves #https://github.com/JoinColony/colonyCDapp/issues/2261

### CODE REVIEW

**Review by melyndav — CHANGES_REQUESTED** [review]
Hey @davecreaser nice start on this one! Just a couple of minor pick ups.

1. The HR between items isn't the right colour token.

image
 
 2. Gap between title and date should be 8px not 6px wide as per figma below. 
 
image

3. When the sidepanel is open and the user selects a transaction etc from the user hub, can we make the user hub close? It currently stays open once the transaction is loaded.

image

**Comment by davecreaser** [comment]
Thanks @melyndav , I've addressed the issues :)

Also just an important note: not all transactions have an associated action, so we can't open the sidebar with an action for those transactions. I've made those transactions not clickable. Hope that makes sense!

File: src/types/transactions.ts [hunk 1]
```diff
@@ -70,3 +70,35 @@ export enum TRANSACTION_METHODS {
   DeployTokenAuthority = 'deployTokenAuthority',
   Approve = 'approve',
 }
+
+export enum BatchKeys {
```
Comment (rdig) [inline src/types/transactions.ts:74]: The batch keys are actually method names, so they should be merged with this: https://github.com/JoinColony/colonyCDapp/blob/185cae2e8073c2241e5425a2a50bd9886e6d68de/src/types/transactions.ts#L67C19-L72

**Review by rdig — APPROVED** [review]
Yup, all good here. Left a comment that needs addressing, but otherwise, nicely done! :100:

**Review by iamsamgibbs — CHANGES_REQUESTED** [review]
Nice work on this ticket, all working nicely and some nice tidying up of the code.

Only one minor issue I've noticed, on the grouped transactions the pills and chevrons should be right aligned (I think they are already, but the rows aren't filling up the entire space, needs min-width: 100% or something).

Screenshot 2024-05-16 at 09 27 13

**Review by melyndav — CHANGES_REQUESTED** [review]
Hey @davecreaser thanks for the fixes.

In the empty state for transactions, could you kindly align the no results to match the Figma design? It's gone a bit awol. 

Same for stakes, could you also align the empty state in the centre of the content block?

image

https://www.figma.com/design/l1dOM5qiQYwF0ElvKDqqjg/Design-System---Colony-v3?node-id=1782-51660&t=YgaJ7uYVBRvSyvEi-4

Also, confirming @iamsamgibbs comment regarding the alignment of the content also needs to be resolved on mobile.

image

Also, within the stakes tab, can you align the pill to the center of the row (consistent with transactions) and also match the font colour of the stakes/ rewards/ penalty text to Figma as well (text-gray-500).

image

**Review by melyndav — CHANGES_REQUESTED** [review]
Hey @davecreaser thanks for your continued work on this!

A couple of refinements are needed in the stakes tab to handle some UI issues.

1. Mobile is displaying the staking data with weird wraps:
image

2. Desktop has some font weight issues and also padding issues with the staking data as well.
image

Font weights should be normal as per figma:
image
 
 There should be 12px padding between each data block as per figma:

image

Also, I think the metadata is showing but in stakes we only want to show stake/reward/penalty data which is contributing to the wrap issue
image

3. There should only be a 8px padding within the transaction accordion to reduce the height of these sections.
image

**Comment by davecreaser** [comment]
Good eagle eyes @melyndav ! Thanks for the time reviewing. I think I've caught everything you've raised now, but let me know if you spot anything else :)

**Review by melyndav — APPROVED** [review]
Super appreciate all the extra work to fix these minor UI bits @davecreaser. Nice work as always 💯

**Pull request merged**